### PR TITLE
OOH - Off of sync States preventing transition of states from business hours to OOH and vice versa

### PR DESCRIPTION
**Pull request by elopezanaya**
### Description
_Include a description of the problem to be solved_

As part of the research for a problem transition states, it was discovered other issues related to OOH

- OOH state related when reloading, was taking the values from cached values instead of from config
- Values from config contains "True", "False", and from cache "true", "false" this was causing problem when evaluating the state
- If when "blue button" was present during already business hours (loaded previously), and then clicked on the button, this displays an error message, but the chat never recovers from that, causing the error pane to always show event after reload , preventing the error pane to show
- When an error happened, header styles where mixing up, causing an OOH header to contain close button, 
- there were situations where a normal session may have an OOH header session, is this was loaded previously
- Transition from OOH to blue button was not possible due to multiple mishaps with states

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

- Load initial states properly
- Transition of states 
 - from OOH to business and being able to start a session
 - from blue to OOH
 - reload after new states should reflect proper states
 - elements like header and button shows styles according to the state
 - Error state is cleaned during reload , to prevent error pane to be shown instead of OOH pane

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

File: chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx [hunk 1]
```diff
@@ -27,20 +27,20 @@ export const ChatButtonStateful = (props: IChatButtonStatefulParams) => {
...
-    
+
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const [state, dispatch]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
     const { buttonProps, outOfOfficeButtonProps, startChat } = props;
     //Setting OutOfOperatingHours Flag
-    const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True");
-    
-    const ref = useRef(() => {return;});
+    const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True" || state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "true");
```
Comment (xTEddie) [inline chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx:35]: Would using `toLower()` make much more sense?

File: chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx [hunk 2]
```diff
@@ -84,10 +84,8 @@ export const ChatButtonStateful = (props: IChatButtonStatefulParams) => {
     };
 
     useEffect(() => {
-        
-        if (state.appStates.outsideOperatingHours) {
-            setOutOfOperatingHours(true);
-        }
+        setOutOfOperatingHours(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True" || state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "true");
```
Comment (xTEddie) [inline chat-widget/src/components/chatbuttonstateful/ChatButtonStateful.tsx:87]: I'd use `toLower()` to simplify condition

File: chat-widget/src/components/headerstateful/HeaderStateful.tsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ export const HeaderStateful = (props: IHeaderStatefulParams) => {
     const [adapter,]: [any, (adapter: any) => void] = useChatAdapterStore();
     const { headerProps, outOfOfficeHeaderProps, endChat } = props;
     //Setting OutOfOperatingHours Flag
-    const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True");
+    const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True" || state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "true");
```
Comment (xTEddie) [inline chat-widget/src/components/headerstateful/HeaderStateful.tsx:37]: I'd use `toLower()` to simplify condition

File: chat-widget/src/components/headerstateful/HeaderStateful.tsx [hunk 4]
```diff
@@ -80,17 +80,17 @@ export const HeaderStateful = (props: IHeaderStatefulParams) => {
...
         hideCloseButton: state.appStates.conversationState === ConversationState.OutOfOffice || outOfOfficeHeaderProps?.controlProps?.hideCloseButton
     };
 
     useEffect(() => {
-        if (state.appStates.outsideOperatingHours) {
-            setOutOfOperatingHours(true);
-        }
-    }, []);
+        console.log("LOPEZ :: HEADER : 1 => ", state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours);
+        setOutOfOperatingHours(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True" || state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "true");
```
Comment (xTEddie) [inline chat-widget/src/components/headerstateful/HeaderStateful.tsx:92]: I'd use `toLower()` to simplify condition

File: chat-widget/src/components/headerstateful/HeaderStateful.tsx [hunk 5]
```diff
@@ -80,17 +80,17 @@ export const HeaderStateful = (props: IHeaderStatefulParams) => {
...
         ...outOfOfficeHeaderProps?.controlProps,
         hideCloseButton: state.appStates.conversationState === ConversationState.OutOfOffice || outOfOfficeHeaderProps?.controlProps?.hideCloseButton
     };
 
     useEffect(() => {
-        if (state.appStates.outsideOperatingHours) {
-            setOutOfOperatingHours(true);
-        }
-    }, []);
+        console.log("LOPEZ :: HEADER : 1 => ", state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours);
```
Comment (xTEddie) [inline chat-widget/src/components/headerstateful/HeaderStateful.tsx:91]: nit: Remove console.log()

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 6]
```diff
@@ -1,11 +1,30 @@
...
+        const liveChatConfig = initialStateFromCache.domainStates?.liveChatConfig?.LiveWSAndLiveChatEngJoin;
+        if (liveChatConfig) {
+            liveChatConfig.OutOfOperatingHours = 
+            initialStateFromCache.appStates.outsideOperatingHours = liveChatConfig.OutOfOperatingHours;
+
+            // this is needed to prevent the case of some errors preventing the ooo pane to show.
+            if (liveChatConfig.OutOfOperatingHours === true) {
+                initialStateFromCache.appStates.conversationState = ConversationState.OutOfOffice;
+            }
+            console.log("Lopez :: SET Operation hours :: config ::", liveChatConfig.OutOfOperatingHours);
```
Comment (xTEddie) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:23]: nit: Remove `console.log()`

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 7]
```diff
@@ -1,11 +1,30 @@
...
+            console.log("Lopez :: SET Operation hours  :: appstate ::", initialStateFromCache.appStates.outsideOperatingHours);
```
Comment (xTEddie) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:24]: nit: Remove `console.log()`

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 8]
```diff
@@ -51,11 +78,15 @@ export const getLiveChatWidgetContextInitialState = (props: ILiveChatWidgetProps
             startChatFailureType: StartChatFailureType.Generic
         },
         appStates: {
-            conversationState: ConversationState.Closed,
+            conversationState: (props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === true ||
```
Comment (xTEddie) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:81]: I'd use `toLower()` to simplify condition

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 9]
```diff
@@ -51,11 +78,15 @@ export const getLiveChatWidgetContextInitialState = (props: ILiveChatWidgetProps
...
-            conversationState: ConversationState.Closed,
+            conversationState: (props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === true || 
+                   props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True")
+                   ? ConversationState.OutOfOffice 
+                   : ConversationState.Closed,
             isMinimized: undefined,
             previousElementIdOnFocusBeforeModalOpen: null,
             startChatFailed: false,
-            outsideOperatingHours: false,
+            outsideOperatingHours: (props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True" ||
```
Comment (xTEddie) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:88]: I'd use `toLower()` to simplify condition

File: chat-widget/src/components/livechatwidget/common/chatDisconnectHelper.ts [hunk 10]
```diff
@@ -27,7 +27,12 @@ const handleChatDisconnect = (props: ILiveChatWidgetProps, state: ILiveChatWidge
             break;
         case false:
             if (hideSendBoxOnConversationEnd !== false) {
-                setWebChatStyles((styles: StyleOptions) => { return { ...styles, hideSendBox: false }; });
+                // this should be enable only for those that dint disable the sandbox
+                if (props?.webChatContainerProps?.webChatStyles?.hideSendBox !== true){
```
Comment (mithunjohny) [inline chat-widget/src/components/livechatwidget/common/chatDisconnectHelper.ts:31]: you already had the other pr to fix this

File: chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts [hunk 11]
```diff
@@ -52,11 +76,13 @@ export const getLiveChatWidgetContextInitialState = (props: ILiveChatWidgetProps
...
         appStates: {
-            conversationState: ConversationState.Closed,
+            conversationState: (props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours.toString().toLowerCase() === "true")
+                ? ConversationState.OutOfOffice
+                : ConversationState.Closed,
             isMinimized: undefined,
             previousElementIdOnFocusBeforeModalOpen: null,
             startChatFailed: false,
-            outsideOperatingHours: false,
+            outsideOperatingHours: (props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours.toString().toLowerCase() === "true"),
```
Comment (mithunjohny) [inline chat-widget/src/contexts/common/LiveChatWidgetContextInitialState.ts:85]: please do the normalizing of props.chatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours.toString().toLowerCase() in OutOfOfficePanePropsBuilder
